lib/active_merchant/billing/gateways/kushki.rb

```diff
@@ -204,6 +206,33 @@ def add_deferred(post, options)
         }
       end
 
+      def add_three_d_secure(post, payment_method, options)
+        three_d_secure = options[:three_d_secure]
+        return unless three_d_secure
+
+        post[:threeDomainSecure] = {}
+        post[:threeDomainSecure][:eci] = three_d_secure[:eci]
+        post[:threeDomainSecure][:specificationVersion] = three_d_secure[:version]
+
+        if payment_method.brand == 'mastercard'
+          post[:threeDomainSecure][:acceptRisk] = three_d_secure[:eci] == '00'
+          post[:threeDomainSecure][:ucaf] = three_d_secure[:cavv]
+          post[:threeDomainSecure][:directoryServerTransactionID] = three_d_secure[:ds_transaction_id] if three_d_secure[:version].to_f > 2.0
```

Comment:
❓ Question:

What should happen if the brand is Mastercard but 3DS version is 1 ?

💬 Comment:

maybe, just to be sure comparison should be `>=` just in case someone sends just 2 and not 2.1 or 2.2

Reply:
Checking tests, 3ds version has no effect in the request or response, statement corrected.